test/extended/openstack/egressip.go

```diff
@@ -0,0 +1,470 @@
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	cloudnetwork "github.com/openshift/client-go/cloudnetwork/clientset/versioned"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	//egressipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+const (
+	egressIPConfigAnnotationKey = "cloud.network.openshift.io/egress-ipconfig"
+	egressAssignableLabelKey    = "k8s.ovn.org/egress-assignable"
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack][egressip] An egressIP", func() {
+
+	var networkClient *gophercloud.ServiceClient
+	var clientSet *kubernetes.Clientset
+	var err error
+	var infraID string
+	var workerNodeList *corev1.NodeList
+	var cloudNetworkClientset cloudnetwork.Interface
+	oc := exutil.NewCLI("openstack")
+
+	g.BeforeEach(func() {
+
+		g.By("Loading the kubernetes clientset")
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		networkClient, err = client(serviceNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Error creating an openstack network client")
+
+		g.By("Getting the network type")
+		networkType, err := getNetworkType(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if networkType != NetworkTypeOVNKubernetes {
+			e2eskipper.Skipf("Test not applicable for '%s' NetworkType (only valid for '%s')", networkType, NetworkTypeOVNKubernetes)
+		}
+
+		infrastructure, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		infraID = infrastructure.Status.InfrastructureName
+
+		g.By("Getting the worker node list")
+		workerNodeList, err = clientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "node-role.kubernetes.io/worker",
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("attached to a floating IP should be kept after EgressIP node failover with OVN-Kubernetes NetworkType", func() {
+		minWorkerNum := 2
+
+		// Skip based on number of required worker nodes
+		g.By(fmt.Sprintf("Check the number of worker nodes (should be %d at least)", minWorkerNum))
+		e2e.Logf("Detected number of workers: '%d'", len(workerNodeList.Items))
+		if len(workerNodeList.Items) < minWorkerNum {
+			e2eskipper.Skipf("Skipping test as there are '%d' worker node(s) and minimum required is '%d'", len(workerNodeList.Items), minWorkerNum)
+		}
+
+		e2e.Logf("Getting 2 workers for the EgressIP node failover checks, aka primary and secondary workers")
+		primaryWorker := workerNodeList.Items[0]
+		secondaryWorker := workerNodeList.Items[1]
```

Comment:
Do we have any tests creating Machines that will be gone later? If so we should have some way to exclude them here.

Reply:
afaik, we have tests creating bogus machines that never creates a node. In that case it does not affect.